### PR TITLE
Added the WITH command used in the CTE syntax.

### PR DESCRIPTION
**Pull request by NickPaul41**
A common table expression (CTE), is a temporary named result set created from a simple SELECT statement that can be used in a subsequent SELECT statement. Each SQL CTE is like a named query, whose result is stored in a virtual table (a CTE) to be referenced later in the main query.

### CODE REVIEW

File: src/main/java/org/servantscode/commons/search/QueryBuilder.java [hunk 1]
```diff
@@ -33,6 +33,15 @@ private enum BuilderState {START, SELECT, FROM, JOIN, WHERE, GROUP, SORT, LIMIT,
...
 
+    public QueryBuilder withCte(String... with){
+        return withCte(asList(with));
+    }
+
+    public QueryBuilder withCte(List<String> with) {
+        setState(BuilderState.WITH_CTE);
+        this.with.addAll(with);
+        return this;
+    }
```
Comment (GregLeitheiser) [inline src/main/java/org/servantscode/commons/search/QueryBuilder.java:45]: This form doesn't allow for parameters to be provided. Is that intentional?

File: src/main/java/org/servantscode/commons/search/QueryBuilder.java [hunk 2]
```diff
@@ -175,6 +184,10 @@ public String getSql() {
         }
 
         StringBuilder sql = new StringBuilder();
+        if(!with.isEmpty())
+            sql.append("WITH ")
+                    .append(String.join(" ", with))
+                    .append(" ");
```
Comment (GregLeitheiser) [inline src/main/java/org/servantscode/commons/search/QueryBuilder.java]: Should each of the provided "WITH" statements provided start with "WITH" (i.e. is the intention that each could add an additional dynamic table structure?

File: src/main/java/org/servantscode/commons/search/QueryBuilder.java (hunk 1 above)
Comment (NickPaul41) [reply src/main/java/org/servantscode/commons/search/QueryBuilder.java:45]: I've update this with the latest commits on this branch.

File: src/main/java/org/servantscode/commons/search/QueryBuilder.java (hunk 2 above)
Comment (NickPaul41) [reply src/main/java/org/servantscode/commons/search/QueryBuilder.java]: I've update this with the latest commits on this branch.

**Pull request merged**